plugins/core/rendering-info/helpers.js

```diff
@@ -171,11 +188,19 @@ function getCompiledToolRuntimeConfig(
     toolRuntimeConfig.id = item._id;
   }
 
+  // if the target defines tool runtime config, apply it
+  if (targetToolRuntimeConfig) {
+    toolRuntimeConfig = Object.assign(
+      toolRuntimeConfig,
+      targetToolRuntimeConfig
+    );
```

Comment:
Can't we just call Object.assign once?
The many conditions which don't do anything, are a bit confusing.
```js
return Object.assign(
  toolRuntimeConfig,
  targetToolRuntimeConfig,
  toolEndpointToolRuntimeConfig,
  requestToolRuntimeConfig
)
```

Reply:
yes, didn't want to change too many things.